src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2363,10 +2366,146 @@ void GraphOptimizer::FusePerformedAsScaleShiftAndFakeQuantize(Graph &graph) {
+                                                const VectorDims& reorderOutOrder) {
+    if (transposeOrder.size() != layoutOrder.size() || layoutOrder.size() != reorderInOrder.size() ||
+        reorderInOrder.size() != reorderOutOrder.size()) {
+        return false;
+    }
+
+    // revLayoutOrder - reverse permutation for layoutOrder
+    auto revLayoutOrder = VectorDims(layoutOrder.size());
+    for (size_t i = 0; i < revLayoutOrder.size(); i++) {
+        revLayoutOrder[layoutOrder[i]] = i;
+    }
+
+    // newTransposeOrder - Transpose layout-aware permutation
+    auto newTransposeOrder = VectorDims(transposeOrder.size());
+    for (size_t i = 0; i < newTransposeOrder.size(); i++) {
+        newTransposeOrder[i] = layoutOrder[transposeOrder[revLayoutOrder[i]]];
+    }
+
+    // reorderOrder - Reorder layout-aware permutation
+    auto reorderOrder = VectorDims(reorderOutOrder.size());
+    for (size_t i = 0; i < reorderOrder.size(); i++) {
+        for (size_t j = 0; j < reorderOrder.size(); j++) {
+            if (reorderOutOrder[i] == reorderInOrder[j]) {
+                reorderOrder[i] = j;
+                continue;
+            }
+        }
+    }
+
+    // summaryOrder - resulting Transpose+Reorder permutation
+    auto summaryOrder = VectorDims(transposeOrder.size());
+    for (size_t i = 0; i < summaryOrder.size(); i++) {
+        summaryOrder[i] = reorderOrder[newTransposeOrder[i]];
+    }
+
+    // check that Transpose+Reorder is the identical permutation
+    for (size_t i = 0; i < summaryOrder.size(); i++) {
+        if (summaryOrder[i] != i) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void GraphOptimizer::mergeTransposeReshapeReorder(Graph& graph,
+                                                  const NodePtr& transposeNode,
+                                                  const NodePtr& reshapeNode,
+                                                  const NodePtr& reorderNode,
+                                                  const bool reverseOrder) {
+    const auto& parentNode = reverseOrder ? reorderNode : transposeNode;
+    const auto& childNode = reverseOrder ? transposeNode : reorderNode;
+    auto nodeBeforeSequence = parentNode->getParentEdgeAt(0)->getParent();
+    auto nodeAfterSequence = childNode->getChildEdgeAt(0)->getChild();
+
+    graph.RemoveEdge(transposeNode->getParentEdgeAt(1));
+    if (reshapeNode)
+        graph.RemoveEdge(reshapeNode->getParentEdgeAt(1));
+
+    // to prevent inPlace conflict we must check that the memory reference is unidirectional or
+    // inPlace memory is not used
+    const auto parentInPlace = parentNode->getParentEdgeAt(0)->inPlace(Edge::LOOK_UP);
+    const auto& childEdges = childNode->getChildEdgesAtPort(0);
+    const auto childInPlace = std::any_of(childEdges.begin(), childEdges.end(), [](const EdgePtr& edge) {
+        return edge->inPlace(Edge::LOOK_DOWN);
+    });
+    bool isOptimized = !(parentInPlace && childInPlace);
```

Comment:
We must perform this check before detaching nodes, because we need to analyze the edges which will be removed. So I can't move this check. I added an explanatory comment in the code